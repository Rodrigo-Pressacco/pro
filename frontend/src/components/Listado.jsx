import React, { useState, useEffect } from "react";
import "../styles/Listado.css"
import alojamientos from '../utils/Alojamientos.json'
import Card from './Card';
import axios from 'axios';
import url from "../utils/datosInicio.json"



const Listado = (props) => {


  const [datos, setDatos] = useState([null, false])

  useEffect(() => {
    axios.get(`http://${url.hostUrl}:8080/productos/todos`)
      .then(res => {
        const persons = res.data;
        if (datos[1] === false) {
          setDatos([persons, true])
        }
      })
  })



  let filtro = datos[0];
  if (props.filterCategoryList[1] || props.filterLocationList[1]) {
    if (props.filterCategoryList[1] && props.filterLocationList[1]) {
      filtro = datos[0].filter((e) => e.ciudad.nombre.concat(", ", e.ciudad.pais) === props.filterLocation && e.categoria.titulo === props.filterCategory);
      // console.log(1,props.filterLocation)
    } else if (props.filterCategoryList[1]) {
      filtro = props.filterCategoryList[0]
      // console.log(2,props.filterCategoryList)
    } else if (props.filterLocationList[1]) {
      filtro = props.filterLocationList[0]
      // console.log(3,props.filterLocationList)
    }
  }

  let h3 = '';
  if (filtro === datos[0]) {
    h3 = "Recomendaciones"
  }
  else {
    if (props.filterCategoryList[1] && props.filterLocationList[1]) {
      if (filtro.length === 0) {
        h3 = 'No se encontraron resultados';
      } else {
        h3 = "Resultados para: " + props.filterCategory + " en " + props.filterLocation
      }
    } else if (props.filterCategoryList[1]) {
      h3 = "Resultados para: " + props.filterCategory
    } else if (props.filterLocationList[1]) {
      h3 = "Resultados para alojamientos en: " + props.filterLocation
    }
  }

  const handleDatacard = (data) => {
    props.handleDataCard(data)
  }

  function handleClick(boolean) {
    props.onFilter(boolean)
  }

  return (
    <div className="div-General-h3">
      <div className="titulo-filter" >
        <h3>
          {h3}
        </h3>
        {(props.filterCategoryList[1] || props.filterLocationList[1]) ? <button className="cancelar-filtros" onClick={() => handleClick(true)}>x Quitar filtros</button> : ''}
      </div>
      <div className="contenedor-listado">
        {datos[1] ? filtro.map((e, index) =>
          <Card id={e.id} url={e.imagen.filter(im => im.titulo.includes("Portada"))[0].url_imagen} categoria={e.categoria.titulo} nombre={e.nombre}
            ubicacion={e.ciudad.nombre} descripcion={e.descripcion} caracteristicas={e.caracteristicas}
            e={alojamientos[e.id - 1]} key={index} handleData={handleDatacard} />) : <></>}
      </div>
    </div>
  )
}

export default Listado