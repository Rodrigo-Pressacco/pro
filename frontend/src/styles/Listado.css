.cancelar-filtros{
    background-color: var(--main-color);
    font-style: normal;
    font-weight: bold;
    font-size: 16px;
    line-height: 19px;
    color: #FFFFFF;
    text-align: center;
    margin-right: 20px;
    margin-top: 20px;
    padding: 5px 10px;
    border-radius: 5px;
    border: #fff;
}

.titulo-filter{
    display: flex;
    align-items: center;
    justify-content: space-between;
    
}

.contenedor-listado {
    display: flex;
    flex-direction: row;
    position: relative;
    justify-content: space-between;
    margin-bottom: 58px; /*TODO: Agregar variable de altura de footer*/
    flex-wrap: wrap;
    margin: 10px 10px;
    margin-bottom: 90px;
}

.tarjeta {
    position: absolute;
    color: red;
}


.div-General-h3{
    background-color: #F3F1ED;
}

.div-General-h3 h3{
    padding: 20px 20px 0;
    font-size: 24px;
    color: #383B58;
}

@media (max-width: 800px) {
    .contenedor-listado {
        flex-direction: column;
        justify-content: space-between;
    }
  
}