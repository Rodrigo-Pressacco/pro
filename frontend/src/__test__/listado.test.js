import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Listado from '../components/Listado'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Listado />', () => {
    let listado
    let useEffect = jest.fn(); 
    
    beforeEach(() => {
        listado = shallow(<Listado useEffect={useEffect} />)
    });

    it("Deberia mostrar <Listado/> correctamente", () => {
        expect(listado).toMatchSnapshot();
    });

})