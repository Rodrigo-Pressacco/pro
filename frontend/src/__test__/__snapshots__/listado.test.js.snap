// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Listado /> Deberia mostrar <Listado/> correctamente 1`] = `
<div
  className="div-General-h3"
>
  <h3>
    Recomendaciones
  </h3>
  <div
    className="contenedor-listado"
  />
</div>
`;
